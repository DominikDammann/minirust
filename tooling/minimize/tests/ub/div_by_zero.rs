#![allow(internal_features)] 
#![feature(core_intrinsics)]
extern crate intrinsics;
use intrinsics::*;

fn main() {
    // Use `unchecked_div` to make div-by-zero UB (rather than panic).
    print(unsafe { std::intrinsics::unchecked_div(1, black_box(0)) });
}

fn black_box<T>(t: T) -> T { t }
