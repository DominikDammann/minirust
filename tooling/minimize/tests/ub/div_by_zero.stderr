error: UB: division by zero

error: aborting due to 1 previous error

